### PR TITLE
MGMT-8200: Fix controller crash when import installed cluster 

### PR DESCRIPTION
**Pull request by filanov**
# Assisted Pull Request

## Description

* Reset API require OpenshiftClusterID in V2ImportClusterInternal but because it is in internal function it can be called without OpenshiftClusterID, for example by controller packages, because this parameter is not required for the ACM we can skip it in the controller.
* Importing cluster can add optional parameter OpenshiftClusterID when creating day2 clsuter without having day1 (import cluster) OpenshiftClusterID is added by setting ACI.spec.clusterMetadata.clusterID
* Add subsystem test that create installed CD and ACI to check that cluster can be imported without being installed by AI controller.

## List all the issues related to this PR

- [ ] New Feature
- [X] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [X] Automation (CI, tools, etc)
- [ ] Cloud
- [X] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [X] Waiting for CI to do a full test run
- [X] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @nmagnezi 
/cc @rollandf 

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by filanov** [comment]
/hold 
waiting for tests to finish

**Comment by filanov** [comment]
/unhold 
local tests passed

**Review by nmagnezi — APPROVED** [review]
/lgtm

File: subsystem/kubeapi_test.go [hunk 1]
```diff
@@ -3277,6 +3277,48 @@ spec:
...
+				Kind:       "AgentClusterInstall",
+				APIVersion: "hiveextension/v1beta1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: Options.Namespace,
+				Name:      clusterDeploymentSpec.ClusterInstallRef.Name,
+			},
+			Spec: *aciSpec,
+			Status: hiveext.AgentClusterInstallStatus{
+				Conditions: []hivev1.ClusterInstallCondition{
```
Comment (rollandf) [inline subsystem/kubeapi_test.go:3298]: Is it needed?
The conditions should be populated by controller, not by the user

Comment (filanov) [reply subsystem/kubeapi_test.go:3298]: I agree that it doesn't make sense to ask from the user to set conditions but for now this is what we have in the code and it's out of scope for this ticket, i added this topic for our meeting tomorrow.
In order for us to create day2 cluster we need:
1. CRD: spec.Installed = ture, Spec.ProvisionRequirements.ControlPlaneAgents > 1
2. ACI complete conditions = true
The problem here is with 2, i think that if CD is marked as installed we can skip the validation of the conditions.

Comment (filanov) [reply subsystem/kubeapi_test.go:3298]: Opened https://issues.redhat.com/browse/MGMT-8224 to handle it.

Comment (filanov) [reply subsystem/kubeapi_test.go]: Had a talk with Fred, looks like we don't really need to have the condition, fixed the tests.

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Pull request merged**